src/test/ui/resolve/issue-69401-trait-fn-no-body-ty-local.rs

```diff
@@ -2,5 +2,5 @@ fn main() {}
 
 trait Foo {
     fn fn_with_type_named_same_as_local_in_param(b: b);
-    //~^ ERROR cannot find type `b` in this scope [E0412]
+    //~^ ERROR expected type, found local variable `b`
```

Comment:
Could this break existing code? This compiles for example:

```rust
struct Foo { x: i32 }

fn foo(Foo: Foo) {
    let _ = Foo;
}
```

Reply:
This only changes the behavior when the function doesn't have a body (e.g. in a trait, or an `extern` function), so that example you gave specifically wouldn't be affected.

Reply:
er, I guess my theory applies to something like

```rust
struct Foo { x: i32 }

trait Trait {
    fn foo(Foo: Foo);
}
```

which also compiles – I was worried it would break because it seems like the second `b` above is being resolved as a local variable, and in this example that could mean the local variable `Foo` shadowing the type `Foo`. But if you're confident it will work then LGTM